src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -189,7 +189,11 @@ private void InitializeValues(HttpMethod method, Uri? requestUri)
             }
             if ((requestUri != null) && (requestUri.IsAbsoluteUri) && (!HttpUtilities.IsHttpUri(requestUri)))
             {
+#if TARGETS_BROWSER
+                throw new ArgumentException(SR.net_http_client_http_browser_baseaddress_required, nameof(requestUri));
+#else
                 throw new ArgumentException(SR.net_http_client_http_baseaddress_required, nameof(requestUri));
+#endif
```

Comment:
This same change is required in the `RequestUri` getter too, IIUC. And tests needed for that.

Reply:
And is this same change required in [HttpClient:CheckBaseAddress](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L946-L949) too? Tests needed for that too.

Reply:
Messages changed.

Reply:
HttpClient can not be tested as it will not run in a non browser environment right now.  This will be tested when we get the https://github.com/dotnet/runtime/tree/master/src/libraries/System.Net.Http/tests running in the browser.